l10n_br_account/models/fiscal_document.py

```diff
@@ -180,3 +183,18 @@ def action_document_back2draft(self):
         if self.move_ids:
             self.move_ids.button_draft()
         return result
+
+    def _exec_after_SITUACAO_EDOC_DENEGADA(self, old_state, new_state):
+        self.ensure_one()
+        models_cancel_on_deny = [MODELO_FISCAL_NFE, MODELO_FISCAL_CTE]
+        if (
+            self.document_type_id.code in models_cancel_on_deny
+            and self.issuer == DOCUMENT_ISSUER_COMPANY
+        ):
+            self._document_deny()
+        return super()._exec_after_SITUACAO_EDOC_DENEGADA(old_state, new_state)
+
+    def _document_deny(self):
+        msg = _("Canceled due to the denial of document %s") % (self.document_number)
```

Comment:
Sugestão para evitar warn no pylint nas próximas versões do Odoo, adicionar placeholder, isso tbm ajuda muito quem tá traduzindo depois:

```suggestion
        msg = _("Canceled due to the denial of document %(document_number)s", 
document_number=self.document_number)

Reply:
eu ia comentar a mesma coisa. Tem outro problema com o hook ligado a extração do l10n_br_fiscal_edi que estamos fazendo antes disso virar tudo um grande spaghetti code que nunca iria ser possível de limpar mais, mas vou propor uma solução mais tarde.